cursive-core/src/views/select_view.rs

```diff
@@ -965,13 +981,16 @@ impl<T: 'static + Send + Sync> View for SelectView<T> {
             printer.with_style(style, |printer| {
                 // Prepare the entire background
                 printer.print_hline((1, 0), x, " ");
-                // Draw the borders
-                printer.print((0, 0), "<");
-                printer.print((x, 0), ">");
+                // Draw the decorators
+                printer.print((0, 0), &self.decorators[0]);
+                printer.print((x + 1 - self.decorators[1].len(), 0), &self.decorators[1]);
 
                 if let Some(label) = self.items.get(focus).map(|item| &item.label) {
                     // And center the text?
-                    let offset = HAlign::Center.get_offset(label.width(), x + 1);
+                    let offset = HAlign::Center.get_offset(
+                        label.width(),
+                        x + 1 - self.decorators.iter().map(|d| d.len()).sum::<usize>(),
```

Comment:
Same here: what we want is the width (in cells), not the length (in bytes).
```suggestion
                        x + 1 - self.decorators.iter().map(|d| d.width()).sum::<usize>(),
```

Reply:
And actually let's use saturating_sub since we don't know how wide the decorators are.
```suggestion
                        (x + 1).saturating_sub(self.decorators.iter().map(|d| d.width()).sum::<usize>()),
```

Reply:
And if you compute `x = printer.size.x.checked_sub(self.decorators[1].width())`, then you only need `x.saturating_sub(self.decorators[0].width())`.